### ISSUE TITLE
Buttons for interaction with Onboardingbot giving 404

### ISSUE THREAD

**Issue by dekadentno** [chiefonboarding/ChiefOnboarding#257]
Hello and thank you for investing time and resources in this great open source project.

I was following your documentation, deployed the app and integrated the slack bot. Everything works well, except that interacting with buttons gives errors, like shown on the picture below: 
 
![Screenshot at 2022-12-05 15-58-55](https://user-images.githubusercontent.com/12261635/205670211-208b6c24-1834-4dc3-961b-a44f2ee95b5f.png)

My web app logs show nothing else except:
`│ 10.233.73.3 - - [05/Dec/2022:15:04:51 +0000] "POST /api/slack/bot HTTP/1.1" 404 6095 "-" "Slackbot 1.0 (+https://api.slack.com/robots)"                                                    │
`

Somehow, while poking around, I've stumbeled upon an error visible in the app settings on slack:
`settings.event_subscriptions.request_url URL isn’t verified. [Learn more](https://api.slack.com/events/url_verification)
https://my-app-url/api/slack/bot — Verification unsuccessful, try again`

IIRC, button click triggers a POST request on `https://my-app-url/api/slack/bot` which is for some reason not found. I have no idea how to resolve this. No luck in the documentation, nor on github issues.

This is a fantastic piece of software that deserves more users. Thank you once again for making it work.

edit: I also found this and I'm not sure what to do:
![Screenshot at 2022-12-05 16-11-59](https://user-images.githubusercontent.com/12261635/205672521-bea3a0f5-55d9-485f-b406-0a62a26ae600.png)

**Comment by GDay** [comment]
Thanks for reporting this! 

For me, all buttons work, however when I try to change the request url to the same url, it won't verify with the challenge for me either.

Without that verification (which it only needs to do once), buttons won't work. You are on the right track. I am not entirely sure yet what caused this issue, but I am looking into it.

Will get back to this asap.

**Comment by GDay** [comment]
Oh, I was wrong. I was using a different (read: old) bot and therefore the token/signing secret didn't work and it wouldn't verify as those didn't match.

It does work for me, without any issues. So, let's debug what happens:

If you try to make that "challenge" call: what status code does it return in the server logs? is it a 401? 

If that's the case, then your signing secret has probably not been filled in correctly. Please do this:

Go into the environment where your instance is running, so you can command line access. Then go into shell:

```
python back/manage.py shell
```
Then get the Slack integration:

```
from admin.integrations.models import Integration
i = Integration.objects.get(integration=0)
```
Then print the secret signing key and check if it matches with what is in your Slack bot config:
```
i.signing_secret
```
If the signing secret is not correct, then you can do this to fix it:
```
i.signing_secret = 'YOURCORRECTSIGNINGKEY'
i.save()
```

Let me know if that worked.

**Comment by dekadentno** [comment]
Thank you for your swift answer @GDay, I'll try to use your suggestions and report back shortly.

**Comment by dekadentno** [comment]
@GDay 

It seems that setting the correct key helped.

Just a note: 
We run it in a docker container inside Kubernetes. We rewrote the docker compose from your git to a kubernetes deployment.
We used the offical docker image from docker hub and `manage.py` was located in `/app/manage.py`
`i.save() ` did not seem to work right away and I guess the container just had to be restarted for the app to reload the new token.

Thank you for your help once again
